src/modules/Form/index.tsx

```diff
@@ -1,210 +1,140 @@
+    setPokemons([...pokemons, customPokemon]);
+    reset(DEFAULT_VALUES, { keepValues: false });
   };
 
-  handleSubmit = (event: FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    const fields = [
-      this.nameField,
-      this.typeField,
-      this.avatarField,
-      this.isShinyField,
-      this.consentField,
-      this.birthdayField,
-      this.maleInputRef,
-      this.femaleInputRef,
-    ];
-    const errors = this.validate(fields);
-    if (Object.values(errors).some((error) => error != null)) {
-      this.setState({ errors });
-    } else {
-      const customPokemon = this.prepareCustomPokemon();
-      this.setState({
-        pokemons: [...this.state.pokemons, customPokemon],
-      });
-      fields.forEach((field) => {
-        const current = field.current;
-        if (current) {
-          if (current.type === 'radio' || current.type === 'checkbox') {
-            (current as HTMLInputElement).checked = false;
-          } else current.value = '';
-        }
-      });
-    }
-  };
+  watch(() => initialEnter && setInitialEnter(false));
 
-  render() {
-    const { location, componentName } = this.props;
-    return (
-      <Layout location={location} componentName={componentName}>
-        <S.CommonView>
-          <S.FormHeading>Create custom pokemon!</S.FormHeading>
-          <S.Form onSubmit={this.handleSubmit}>
-            <NameField
-              onChange={this.removeError}
-              errors={this.state.errors}
-              ref={this.nameField}
-            />
-            <TypeField
-              onChange={this.removeError}
-              errors={this.state.errors}
-              ref={this.typeField}
-            />
-            <S.GenderWrapper>
-              <S.RadioWrapper>
-                *Gender
-                <S.RadioFields>
-                  <GenderField
-                    name={FEMALE}
-                    ref={this.femaleInputRef}
-                    onChange={this.removeError}
-                  />
-                  <GenderField name={MALE} ref={this.maleInputRef} onChange={this.removeError} />
-                </S.RadioFields>
-              </S.RadioWrapper>
-              <ErrorMessage
-                visible={!!this.state.errors.gender}
-                message={this.state.errors.gender}
-              />
-            </S.GenderWrapper>
-            <BirthdayField
-              ref={this.birthdayField}
-              errors={this.state.errors}
-              onChange={this.removeError}
-            />
-            <AvatarField ref={this.avatarField} />
-            <ShinyField ref={this.isShinyField} />
-            <ConsentField
-              ref={this.consentField}
-              errors={this.state.errors}
-              onChange={this.removeError}
+  const onChange = (field: Fields) => ({
+    onChange: () => {
+      !!errors[field]?.message && clearErrors(field);
+      !!showSuccessMessage && setShowSuccessMessage(false);
+    },
+  });
```

Comment:
we are returning object with function? Are you sure it is correct? Bad readability. Is it lib specific syntax?

Reply:
Yes, it's specific for lib syntax. Input register function takes name and optional object in which we can declare onChange. 
So to not repeat too much code I created a function that prepares register options with onChange method. I agree that it should be named better.

To ilustrate the problem:

With prepare function:    {...register(Fields.name, onChange(Fields.name))}
Without prepare function:  {...register(Fields.name, {
      onChange: () => {
          !!errors[Fields.name]?.message && clearErrors(Fields.name);
          !!showSuccessMessage && setShowSuccessMessage(false);
     },
 }

And for each input only Field name changes

Reply:
I compared few ways of improving readability and I think the best is:

const onChange = (field: Fields) => {
    !!errors[field]?.message && clearErrors(field);
    !!showSuccessMessage && setShowSuccessMessage(false);
};

And register:

{...register(Fields.type, { onChange: () => onChange(Fields.type) }}